src/main/java/com/coherent/training/selenium/kapitsa/web/providers/ConfigFileReader.java

```diff
@@ -27,14 +54,28 @@ public static ConfigFileReader getInstance(){
         return SingletonConfigFileReader.INSTANCE;
     }
 
-    public String getHubURL() {
-        String hubURL = properties.getProperty(HUB_URL.getPropertyKey());
+    public String getSeleniumHubURL() {
+        properties = getPropertyFromCache(PROFILE_PATH);
+        String profileKey = String.format(SELENIUM_URL_KEY.getPropertyKey(), "HUB");
+
+        String hubURL = properties.getProperty(profileKey);
+
+        if (hubURL != null) return hubURL;
+        else throw new RuntimeException("selenium hubURL is not specified");
+    }
+
+    public String getSeleniumSauceLabURL(){
+        properties = getPropertyFromCache(PROFILE_PATH);
+        String profileKey = String.format(SELENIUM_URL_KEY.getPropertyKey(), "SAUCELAB");
+
```

Comment:
you can unify these two methods into one - getSeleniumUrl
String profileKey = String.format(SELENIUM_URL_KEY.getPropertyKey(), getProperty(PROFILE));